From d41790ca28728a31cc9a178e633cf6dac508e42a Mon Sep 17 00:00:00 2001
From: Yonghua Zheng <yonghua.zheng@aeroratech.com>
Date: Fri, 18 Oct 2024 13:35:06 +0800
Subject: [PATCH] remove Requires.private from pkgconfig

Requires.private lists the dependencies required for the library
and not for the program.

+----------------+   +----------------+   +---------------+
| program        |-->| your lib       |-->| required lib  |
+----------------+   +----------------+   +---------------+

However in Yocto pkg-config reports dependencies as listed in
Requires.private and causes build issues. Remove it for now
to fix that.

diff --git a/src/mavsdk.pc.in b/src/mavsdk.pc.in
index 320c33d1..b21f2894 100644
--- a/src/mavsdk.pc.in
+++ b/src/mavsdk.pc.in
@@ -6,6 +6,5 @@ includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 Name: MAVSDK
 Description: API and library for MAVLink compatible systems written in C++17
 Version: @MAVSDK_VERSION_STRING@
-Requires.private: libcurl jsoncpp tinyxml2
 Libs: -L"${libdir}" -lmavsdk
 Cflags: -I"${includedir}" -I"${includedir}/mavsdk"
-- 
2.34.1

